server/cmd/server.go

```diff
@@ -210,7 +211,24 @@ func Run(ctx context.Context, opts options.CompletedOptions) error {
 		return err
 	}
 
-	return prepared.Run(ctx)
+	// Run the server and wait for readiness
+
+	go func() {
+		if err := prepared.Run(ctx); err != nil {
+			klog.Fatal(err, "Failed to run server")
+		}
+	}()
+
+	// wait for the server to be ready
+	klog.Info("Waiting for control plane to be ready")
+	err = readiness.WaitForReady(ctx, completed.Options.AdminAuthentication.KubeConfigPath)
```

Comment:
is this just for cosmetics, to have nice output?

Reply:
Right, to log out when something is not yet ready, but in reality control-plane is semi-functional already.

Reply:
This was my mistake. Perception was that its "healthy" but something was not starting right.